contracts/contracts/PoolDeposits.sol

```diff
@@ -217,12 +223,30 @@ contract PoolDeposits {
   /// @dev Lets a user withdraw their original amount sent to DAOcare
   /// Calls the NoLossDao conrrtact to determine eligibility to withdraw
   /// Withdraws the proposalAmount (50DAI) if succesful
-  function withdrawDeposit() external userStaked {
-    _withdrawFunds();
+  function exit() external userStaked {
+    uint256 amount = depositedDai[msg.sender];
+    // NOTE: it is critical that _removeEmergancyVote happens before _withdrawFunds.
+    _removeEmergencyVote();
+    _withdrawFunds(amount);
     noLossDaoContract.noLossWithdraw(msg.sender);
     emit DepositWithdrawn(msg.sender);
   }
 
+  /// @dev Lets a user withdraw their original amount sent to DAOcare
+  /// Calls the NoLossDao conrrtact to determine eligibility to withdraw
+  /// Withdraws the proposalAmount (50DAI) if succesful
+  function withdrawDeposit(uint256 amount)
```

Comment:
Conceptually this function should be the same as the user doing 2 transactions - 1 to exit, another to join with less deposit. (Similar goes for adding deposit).

The user should be given no benefit to running this single function over just calling those 2 functions separately besides convenience and gas savings.

Reply:
Also there are ZERO tests for this function yet (But existing tests all pass)

Reply:
Got it. We will make sure our coverage is back at 100%

Reply:
Here is the issue, nowhere do this function check if you have voted in the DAO. Why not join the dao on 11hr with 1000000 DAI, vote 10 seconds later in the new iteration, then withdraw 999999DAI and have your vote stand?

The exit function calls noLossDaoContract.noLossWithdraw(msg.sender); and the other contract checks this logic. 

This is the only major issue I've spotted thats need fixing.

We need to call the dao contract and check if the user has voted, if so, reject the partial withdrawl.